hack/tools/go.mod

```diff
@@ -27,7 +27,7 @@ require (
 	sigs.k8s.io/cluster-api v0.0.0-00010101000000-000000000000
 	sigs.k8s.io/cluster-api/test v0.0.0-00010101000000-000000000000
 	sigs.k8s.io/controller-runtime v0.20.4
-	sigs.k8s.io/controller-tools v0.17.0
+	sigs.k8s.io/controller-tools v0.17.3
```

Comment:
Let's keep this on 0.17.3 in this PR and not bump to 0.18 like I suggested

Bumping this to v0.18 would require CR v0.21, but just bumping the controller-gen binary is fine

I'll bump this to 0.18 as part of my CR PR